x-pack/plugins/security_solution/common/api/timeline/patch_timelines/patch_timeline_route.gen.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from 'zod';
+
+/*
+ * NOTICE: Do not edit this file manually.
+ * This file is automatically generated by the OpenAPI Generator, @kbn/openapi-generator.
+ */
+
+import { SavedTimeline, TimelineResponse } from '../model/components.gen';
+
+export type PatchTimelineRequestBody = z.infer<typeof PatchTimelineRequestBody>;
+export const PatchTimelineRequestBody = z.object({
+  timelineId: z.string().optional(),
+  version: z.string().optional(),
+  timeline: SavedTimeline.optional(),
```

Comment:
Are all of these optional? I would assume a request that does not contain all three of these to fail.

Reply:
Yea thanks. The openApi schema defaults to optional, and you have to include all fields that are required individually 😮‍💨 . Anyways, updated. Thanks for the note!